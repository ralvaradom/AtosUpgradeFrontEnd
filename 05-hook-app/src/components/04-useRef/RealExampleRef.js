import React, { useState } from 'react';
import '../02-useEffect/effect.css';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';

const RealExampleRef = () => {

	const [ show, setShow ] = useState(false);

  	return (
  		<div>
				<h1>Real Example Ref</h1>
				{ show && <MultipleCustomHooks /> }
				<button
					className="btn btn-primary mt-5"
					onClick={() => setShow(!show)}
				>Show/Hide
				</button>
		</div>
	);
};

export default RealExampleRef;
