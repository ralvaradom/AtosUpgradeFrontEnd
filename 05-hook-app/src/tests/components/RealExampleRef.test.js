import React from 'react';
import { shallow } from 'enzyme';
import RealExampleRef from '../../components/04-useRef/RealExampleRef';


describe('Pruebas en <MultileCustomHooks />', () => {

	const wrapper = shallow(<RealExampleRef />);
	
	test('debe mostrarse correctamente', () => {

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
	});

	test('debe de mostrar la informacion', () => {

		wrapper.find('button').simulate('click');

		expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
	});
});
