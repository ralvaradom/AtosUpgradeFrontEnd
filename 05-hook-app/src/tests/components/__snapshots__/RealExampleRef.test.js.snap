// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultileCustomHooks /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
